// tw-element/es/components/upload/src/upload-list.mjs
import { defineComponent, ref, resolveComponent, openBlock, createBlock, TransitionGroup, normalizeClass, withCtx, createElementBlock, Fragment, renderList, withKeys, renderSlot, createCommentVNode, createElementVNode, createVNode, createTextVNode, toDisplayString } from 'vue';
import { NOOP } from '@vue/shared';
import { TwIcon } from '../../icon/index.mjs';
import { Document, Delete, Close, ZoomIn, Check, CircleCheck } from '@element-plus/icons-vue';
import '../../../hooks/index.mjs';
import { TwProgress } from '../../progress/index.mjs';
import _export_sfc from '../../../_virtual/plugin-vue_export-helper.mjs';
import { useLocale } from '../../../hooks/use-locale/index.mjs';
import { useNamespace } from '../../../hooks/use-namespace/index.mjs';

const _sfc_main = defineComponent({
  name: "TwUploadList",
  components: {
    TwProgress,
    TwIcon,
    Document,
    Delete,
    Close,
    ZoomIn,
    Check,
    CircleCheck
  },
  props: {
    files: {
      type: Array,
      default: () => []
    },
    disabled: {
      type: Boolean,
      default: false
    },
    handlePreview: {
      type: Function,
      default: () => NOOP
    },
    listType: {
      type: String,
      default: "text"
    }
  },
  emits: ["remove"],
  setup(props, { emit }) {
    const { t } = useLocale();
    const nsUpload = useNamespace("upload");
    const nsIcon = useNamespace("icon");
    const nsList = useNamespace("list");
    const handleClick = (file) => {
      props.handlePreview(file);
    };
    const onFileClicked = (e) => {
      ;
      e.target.focus();
    };
    const handleRemove = (file) => {
      emit("remove", file);
    };
    return {
      focusing: ref(false),
      handleClick,
      handleRemove,
      onFileClicked,
      t,
      nsUpload,
      nsIcon,
      nsList
    };
  }
});
const _hoisted_1 = ["onKeydown"];
const _hoisted_2 = ["src"];
const _hoisted_3 = ["onClick"];
const _hoisted_4 = ["onClick"];
const _hoisted_5 = ["onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_document = resolveComponent("document");
  const _component_el_icon = resolveComponent("el-icon");
  const _component_circle_check = resolveComponent("circle-check");
  const _component_check = resolveComponent("check");
  const _component_close = resolveComponent("close");
  const _component_el_progress = resolveComponent("el-progress");
  const _component_zoom_in = resolveComponent("zoom-in");
  const _component_delete = resolveComponent("delete");
  return openBlock(), createBlock(TransitionGroup, {
    tag: "ul",
    class: normalizeClass([
      _ctx.nsUpload.b("list"),
      _ctx.nsUpload.bm("list", _ctx.listType),
      _ctx.nsUpload.is("disabled", _ctx.disabled)
    ]),
    name: _ctx.nsList.b()
  }, {
    default: withCtx(() => [
      (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.files, (file) => {
        return openBlock(), createElementBlock("li", {
          key: file.uid || file,
          class: normalizeClass([
            _ctx.nsUpload.be("list", "item"),
            _ctx.nsUpload.is(file.status),
            { focusing: _ctx.focusing }
          ]),
          tabindex: "0",
          onKeydown: withKeys(($event) => !_ctx.disabled && _ctx.handleRemove(file), ["delete"]),
          onFocus: _cache[0] || (_cache[0] = ($event) => _ctx.focusing = true),
          onBlur: _cache[1] || (_cache[1] = ($event) => _ctx.focusing = false),
          onClick: _cache[2] || (_cache[2] = (...args) => _ctx.onFileClicked && _ctx.onFileClicked(...args))
        }, [
          renderSlot(_ctx.$slots, "default", { file }, () => [
            file.status !== "uploading" && ["picture-card", "picture"].includes(_ctx.listType) ? (openBlock(), createElementBlock("img", {
              key: 0,
              class: normalizeClass(_ctx.nsUpload.be("list", "item-thumbnail")),
              src: file.url,
              alt: ""
            }, null, 10, _hoisted_2)) : createCommentVNode("v-if", true),
            createElementVNode("a", {
              class: normalizeClass(_ctx.nsUpload.be("list", "item-name")),
              onClick: ($event) => _ctx.handleClick(file)
            }, [
              createVNode(_component_el_icon, {
                class: normalizeClass(_ctx.nsIcon.m("document"))
              }, {
                default: withCtx(() => [
                  createVNode(_component_document)
                ]),
                _: 1
              }, 8, ["class"]),
              createTextVNode(" " + toDisplayString(file.name), 1)
            ], 10, _hoisted_3),
            createElementVNode("label", {
              class: normalizeClass(_ctx.nsUpload.be("list", "item-status-label"))
            }, [
              _ctx.listType === "text" ? (openBlock(), createBlock(_component_el_icon, {
                key: 0,
                class: normalizeClass([_ctx.nsIcon.m("upload-success"), _ctx.nsIcon.m("circle-check")])
              }, {
                default: withCtx(() => [
                  createVNode(_component_circle_check)
                ]),
                _: 1
              }, 8, ["class"])) : ["picture-card", "picture"].includes(_ctx.listType) ? (openBlock(), createBlock(_component_el_icon, {
                key: 1,
                class: normalizeClass([_ctx.nsIcon.m("upload-success"), _ctx.nsIcon.m("check")])
              }, {
                default: withCtx(() => [
                  createVNode(_component_check)
                ]),
                _: 1
              }, 8, ["class"])) : createCommentVNode("v-if", true)
            ], 2),
            !_ctx.disabled ? (openBlock(), createBlock(_component_el_icon, {
              key: 1,
              class: normalizeClass(_ctx.nsIcon.m("close")),
              onClick: ($event) => _ctx.handleRemove(file)
            }, {
              default: withCtx(() => [
                createVNode(_component_close)
              ]),
              _: 2
            }, 1032, ["class", "onClick"])) : createCommentVNode("v-if", true),
            createCommentVNode(" Due to close btn only appears when li gets focused disappears after li gets blurred, thus keyboard navigation can never reach close btn"),
            createCommentVNode(" This is a bug which needs to be fixed "),
            createCommentVNode(" TODO: Fix the incorrect navigation interaction "),
            !_ctx.disabled ? (openBlock(), createElementBlock("i", {
              key: 2,
              class: normalizeClass(_ctx.nsIcon.m("close-tip"))
            }, toDisplayString(_ctx.t("el.upload.deleteTip")), 3)) : createCommentVNode("v-if", true),
            file.status === "uploading" ? (openBlock(), createBlock(_component_el_progress, {
              key: 3,
              type: _ctx.listType === "picture-card" ? "circle" : "line",
              "stroke-width": _ctx.listType === "picture-card" ? 6 : 2,
              percentage: +file.percentage,
              style: { "margin-top": "0.5rem" }
            }, null, 8, ["type", "stroke-width", "percentage"])) : createCommentVNode("v-if", true),
            _ctx.listType === "picture-card" ? (openBlock(), createElementBlock("span", {
              key: 4,
              class: normalizeClass(_ctx.nsUpload.be("list", "item-actions"))
            }, [
              createElementVNode("span", {
                class: normalizeClass(_ctx.nsUpload.be("list", "item-preview")),
                onClick: ($event) => _ctx.handlePreview(file)
              }, [
                createVNode(_component_el_icon, {
                  class: normalizeClass(_ctx.nsIcon.m("zoom-in"))
                }, {
                  default: withCtx(() => [
                    createVNode(_component_zoom_in)
                  ]),
                  _: 1
                }, 8, ["class"])
              ], 10, _hoisted_4),
              !_ctx.disabled ? (openBlock(), createElementBlock("span", {
                key: 0,
                class: normalizeClass(_ctx.nsUpload.be("list", "item-delete")),
                onClick: ($event) => _ctx.handleRemove(file)
              }, [
                createVNode(_component_el_icon, {
                  class: normalizeClass(_ctx.nsIcon.m("delete"))
                }, {
                  default: withCtx(() => [
                    createVNode(_component_delete)
                  ]),
                  _: 1
                }, 8, ["class"])
              ], 10, _hoisted_5)) : createCommentVNode("v-if", true)
            ], 2)) : createCommentVNode("v-if", true)
          ])
        ], 42, _hoisted_1);
      }), 128))
    ]),
    _: 3
  }, 8, ["class", "name"]);
}
var UploadList = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);

export { UploadList as default };
