// tw-element/es/components/popconfirm/src/popconfirm2.mjs
import '../../button/index.mjs';
import { QuestionFilled } from '@element-plus/icons-vue';
import '../../../utils/index.mjs';
import '../../tooltip/index.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props.mjs';
import { buttonType } from '../../button/src/button2.mjs';
import { iconPropType } from '../../../utils/vue/icon.mjs';
import { useTooltipContentProps } from '../../tooltip/src/tooltip.mjs';

const popconfirmProps = buildProps({
  title: {
    type: String
  },
  confirmButtonText: {
    type: String
  },
  cancTwButtonText: {
    type: String
  },
  confirmButtonType: {
    type: String,
    values: buttonType,
    default: "primary"
  },
  cancTwButtonType: {
    type: String,
    values: buttonType,
    default: "text"
  },
  icon: {
    type: iconPropType,
    default: QuestionFilled
  },
  iconColor: {
    type: String,
    default: "#f90"
  },
  hideIcon: {
    type: Boolean,
    default: false
  },
  hideAfter: {
    type: Number,
    default: 200
  },
  onConfirm: {
    type: definePropType(Function)
  },
  onCancel: {
    type: definePropType(Function)
  },
  teleported: useTooltipContentProps.teleported
});

export { popconfirmProps };
