// tw-element/lib/components/col/src/col.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
require('../../../utils/index.js');
require('../../../hooks/index.js');
var props = require('../../../utils/vue/props.js');
var typescript = require('../../../utils/typescript.js');
var index = require('../../../hooks/use-namespace/index.js');

const colProps = props.buildProps({
  tag: {
    type: String,
    default: "div"
  },
  span: {
    type: Number,
    default: 24
  },
  offset: {
    type: Number,
    default: 0
  },
  pull: {
    type: Number,
    default: 0
  },
  push: {
    type: Number,
    default: 0
  },
  xs: {
    type: props.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  sm: {
    type: props.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  md: {
    type: props.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  lg: {
    type: props.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  xl: {
    type: props.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  }
});
var Col = vue.defineComponent({
  name: "TwCol",
  props: colProps,
  setup(props, { slots }) {
    const { gutter } = vue.inject("TwRow", { gutter: { value: 0 } });
    const ns = index.useNamespace("col");
    const style = vue.computed(() => {
      if (gutter.value) {
        return {
          paddingLeft: `${gutter.value / 2}px`,
          paddingRight: `${gutter.value / 2}px`
        };
      }
      return {};
    });
    const classList = vue.computed(() => {
      const classes = [];
      const pos = ["span", "offset", "pull", "push"];
      pos.forEach((prop) => {
        const size = props[prop];
        if (typeof size === "number") {
          if (prop === "span")
            classes.push(ns.b(`${props[prop]}`));
          else if (size > 0)
            classes.push(ns.b(`${prop}-${props[prop]}`));
        }
      });
      const sizes = ["xs", "sm", "md", "lg", "xl"];
      sizes.forEach((size) => {
        if (typeof props[size] === "number") {
          classes.push(ns.b(`${size}-${props[size]}`));
        } else if (typeof props[size] === "object") {
          const sizeProps = props[size];
          Object.keys(sizeProps).forEach((prop) => {
            classes.push(prop !== "span" ? ns.b(`${size}-${prop}-${sizeProps[prop]}`) : ns.b(`${size}-${sizeProps[prop]}`));
          });
        }
      });
      if (gutter.value) {
        classes.push("is-guttered");
      }
      return classes;
    });
    return () => vue.h(props.tag, {
      class: [ns.b(), classList.value],
      style: style.value
    }, [vue.renderSlot(slots, "default")]);
  }
});

exports.colProps = colProps;
exports["default"] = Col;
