// tw-element/lib/components/menu/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../utils/index.js');
var menu = require('./src/menu.js');
var menuItem$1 = require('./src/menu-item.js');
var menuItemGroup$1 = require('./src/menu-item-group2.js');
var subMenu = require('./src/sub-menu.js');
var menuItem = require('./src/menu-item2.js');
var menuItemGroup = require('./src/menu-item-group.js');
require('./src/types.js');
var install = require('../../utils/vue/install.js');

const TwMenu = install.withInstall(menu["default"], {
  MenuItem: menuItem$1["default"],
  MenuItemGroup: menuItemGroup$1["default"],
  SubMenu: subMenu["default"]
});
const TwMenuItem = install.withNoopInstall(menuItem$1["default"]);
const TwMenuItemGroup = install.withNoopInstall(menuItemGroup$1["default"]);
const TwSubMenu = install.withNoopInstall(subMenu["default"]);

exports.menuEmits = menu.menuEmits;
exports.menuProps = menu.menuProps;
exports.subMenuProps = subMenu.subMenuProps;
exports.menuItemEmits = menuItem.menuItemEmits;
exports.menuItemProps = menuItem.menuItemProps;
exports.menuItemGroupProps = menuItemGroup.menuItemGroupProps;
exports.TwMenu = TwMenu;
exports.TwMenuItem = TwMenuItem;
exports.TwMenuItemGroup = TwMenuItemGroup;
exports.TwSubMenu = TwSubMenu;
exports["default"] = TwMenu;
